### PR TITLE
Fix sentences not ending at end of line

### PR DESCRIPTION
**Pull request by richsalz**
As discussed in https://github.com/openssl/openssl/issues/9593 and some comments in #9604 

This is currently a WIP, to get review of the scripts

### CODE REVIEW

File: util/find-doc-nits [hunk 1]
```diff
@@ -151,6 +151,23 @@ sub name_synopsis()
...
+{
+    my $id = shift;
+    my $contents = shift;
+
+    $contents =~ s/=head1 DESCRIPTION(.*)=head1 COPYRIGHT.*/$1/ms;
+    foreach my $line ( split /\n+/, $contents ) {
+        if ( $line !~ /^ /) {
+            # Look for "period space" if not proceeded by i.e. or e.g.
+            print $id, "Linebreak needed:\n  $line\n"
+                if $line =~ /(?<!([Ii]\.e|[Ee]\.g))\. +/;
```
Comment (levitte) [inline util/find-doc-nits]: I was wondering why not `\s`, but then realised you might not want to match vertical space characters. Tip: `\h` matches horizontal space characters.

Comment (richsalz) [reply util/find-doc-nits]: We don't have tabs and the podpages are ascii, so I think a literal space character is the right thing.

Comment (mspncp) [reply util/find-doc-nits]: Wouldn't a case insensitive regular expression be a bit more readable? `/(?<!(i\.e|e\.g))\. +/i`. 

_(Even if there was an abbreviation like "i.E." (which actually exists in German), the exception would still apply.)_

**Comment by paulidale** [comment]
I'm wondering what the point of this PR is.
Does one space vs two spaces justify a change that will make back porting difficult?

**Comment by levitte** [comment]
> I'm wondering what the point of this PR is.

To be the end all of the one vs two space discussion, which tends to come back again, and again, and again...

Also, I'd say that this change is applicable to 1.1.1 as well, so the difficulty of backporting, at least from *our* perspective, should be rather small...

**Comment by paulidale** [comment]
Despite the endless discussion, I still doubt the usefulness of this change.  Personally, I find writing documentation where every sentence needs to start on a new line quite a bit more difficult than pressing space(s) and letting the text flow.

People who whine about one vs two spaces after a full stop are free to filter the documentation through a sed/perl script.

File: util/find-doc-nits (hunk 1 above)
Comment (richsalz) [reply util/find-doc-nits]: sure, looks more readable, commit pushed.

**Comment by richsalz** [comment]
I agree that breaking lines after each period can disrupt the writing flow.  It seems to be a growing trend for docs, however.  Some links: https://sembr.org
https://rhodesmill.org/brandon/2012/one-sentence-per-line/
https://www.gnu.org/software/emacs/manual/html_node/emacs/Visual-Line-Mode.html  https://www.cultofpedagogy.com/two-spaces-after-period/ It's also the style for FreeBSD manpages, probably others.

**Comment by levitte** [comment]
The reason I have pushed for line breaks at the end of the sentence is purely technical. Certain \*roff renderers only detect periods and similar when they find it at the end of a line, and can then apply *local standards* re spacing.

I have no issues abolishing this habit. However, I will not listen to repeats of the spacing dispute every time I write text.

**Comment by richsalz** [comment]
I added a "-b" flag.  This *temporary measure* turns off the linebreak-checking. This is now out of WIP and ready for review.  If merged, I will then run the utility scripts to fix the linebreaks and create a PR (and backport that PR as well).

**Comment by richsalz** [comment]
Any decision from the OMC?  In the spirit of https://github.com/openssl/openssl/pull/9333#issuecomment-528781239, this can be merged now and then the -b switch removed (b for blanks after periods) and the fix-up script run later on.

**Comment by richsalz** [comment]
Any result from the OMC?  It's been three-plus weeks.  Note that right now this PR doesn't actually complain about the policy, it's just the infrastruture for it.

**Comment by mattcaswell** [comment]
I personally don't feel any need for a new line at the end of each sentence, and agree with @paulidale's comment above.

**Comment by levitte** [comment]
I'm really quite neutral to this, so if everyone else agrees not to care, I say fine.

I guess that means this is getting closed, @richsalz?

**Comment by richsalz** [comment]
Before closing, here's a question: Do you want find-doc-nits to warn about ```.  ``` that is, *two* spaces?  We should be consistent.

**Comment by levitte** [comment]
No. If we don't care, we don't care.

**Pull request closed without merging**